common/plugin_control.c

```diff
@@ -501,26 +508,21 @@ static char **str_split(const char *s, int sep, size_t *field_count)
  *   A pointer to the start of the array if successful
  *   NULL otherwise
  */
-static char **csv_split(const char *s, size_t *field_count)
+static char **line_split_and_unescape(const char *s, size_t *field_count)
 {
     size_t n = 1;               /* One more string than separators. */
     size_t len;                 /* Length of 's' */
-    bool in_string = false;
 
     if (s == NULL || *s == '\0') {
         *field_count = 0;
         return (char **)calloc(1, sizeof(char *));
     } else {
         /* Count separators. */
         for (len = 0; s[len]; ++len) {
-            if (s[len] == ',' && !in_string) {
+            if (s[len] == '\\') {
+                ++len;
```

Comment:
If the string ended with a single `\`  then I think we'd run off the end of the string. (Of course the string shouldn't end like that, but perhaps best to be cautious.)

Reply:
Fixed both these, I think.